cmd/conveyord/shutdown.go

```
@@ -0,0 +1,155 @@
+			httpDone = nil
+		case <-drainCtx.Done():
+			goto hardStop
+		}
+	}
+
+hardStop:
+	cancelDrain()
+	if !riverFinished {
+		select {
+		case riverErr = <-riverDone:
+			riverFinished = true
```

golangci-lint (2.13.2)

[error] ineffectual assignment to riverFinished

(ineffassign)